### DISCUSSION TITLE
Questions regarding #3831 for Enno

### DISCUSSION THREAD

**Discussion by cinnion** [cobbler/cobbler#3832] [Q&A]
OK, bringing a discussion from #3831 to here (first ever GH discussion, BTW, after 10+ years, including years for work).

@SchoolGuy, you were wanting me to get you an explicit build for #3831.  Being used to building my RPMs and such directly on the machine, while I am familiar with Docker, I am not that familiar with the project's docker image. Are you wanting me to do the `make rpms`  there, run system testing suite from the testing page there, or what? And what do you want me to capture, such as the output in `script`?

**Comment by SchoolGuy** [comment]
Tldr is to copy the folder for Rocky 8 from the main branch and adapt everything for Rocky 9 and then add this in the PR.

Then we want to duplicate the Rocky 8 block also in the Compose file so we can build everything at once.

Of course we will need a new target for the Makefile as well but this should be a copy & paste action as well.

**Reply by cinnion** [reply]
Sounds good. I figure we will want an EL9 (Rocky 9) folder as well, correct? I will get that back to you ASAP.

**Reply by SchoolGuy** [reply]
Yes exactly.

**Reply by cinnion** [reply]
BTW... rather than calling the directory `Rocky_Linux_8`, since there are more EL versions out there beyond Rocky, what do you think of calling it `Enterprise_Linux_8` or `EL8`?  (Learning so much about the project and the workflows today, probably more than my time going back to like 1.6 or so! 😄)

**Reply by SchoolGuy** [reply]
I am not in favour of this idea since there are small and subtle differences between the different derivates of EL8, since we are not aware of all of them I think the folder name is fitting.

**Comment by cinnion** [comment]
Sorry about that... was wanting a local merge/pull, to verify what I was seeing running act locally on my dev box.  Apparently, act is reporting more errors on my machine than the pull request.  Grrr... can I get a do-over on today?

**Reply by SchoolGuy** [reply]
There are no deadlines. Take all the time you need.

**Comment by cinnion** [comment]
Morning everyone,

Since one of the problems I needed to address for PR 3831 was breaking of the package builds, I wanted to verify that things were working without my PR (which is partly why the PR goof last night). After all, when investigating a possible regression, you check to make sure that it did not occur beforehand. And being the first time I have used nektos/act to run the workflows locally, I was doubting my local results. But after some food and sleep, along with several more runs (with the `--quiet` flag to turn down the verbosity), I am left thinking that the problem is not entirely my PR, but with the release33 branch itself.

I am attaching my latest log, in hopes that someone can double check me on this, especially if for example @SchoolGuy or @opoplawski use act themselves. Maybe it is just me, but maybe not.  Either way, it needs fixed, and while I am 99% a RHEL/CentOS and now Rocky person, I am willing to do so, because we definitely need things working.

Oh... and starting to like the use of containers for this project more and more. While I have used containers in other ways, this is the first time I have really seen them used beyond an actual deployed service or a replacement for a development VM this extensively.

Regards,
- Doug

[act.log](https://github.com/user-attachments/files/17517663/act.log)

**Reply by SchoolGuy** [reply]
So I saw the logs in the CI already. There are two issues:

-  In the PR the files for Rocky 9 are still missing.
- The error in the specfile that causes the different openSUSE Leap and Debian to think they need the - non existant - `fence-agents-all` package.

**Reply by cinnion** [reply]
I have been wondering about this response, given the following:

1.  The log file I attached was off of the `release33` branch, not the branch with my changes, and using nektos/act to run the workflow locally, I still show the breakage, which is why I attached my log file as opposed to relying on the logs from the CI. Apparently, we would have to do a PR to the main branch to run the CI against that branch unless we set things up just right, am I correct?

2. I have been trying to investigate how the Debian package building is driven, looking for an equivalent to the `.spec` file which drives the `rpmbuild` process. Are we somehow using the `.spec` file to build a control file, or whatever it is that runs the `debuild` process, or do we supply the Debian equivalent, and if so, where are they? The fact that I changed only the `.spec` file makes me think the former, and if so, then I would say it is a bug in that program/process on handling the conditional. That, or if somehow if those conditionals copy over, the `dpkg` or `apt` is misprocessing things, behaving like it claims to be a RHEL release, which it clearly is not, since otherwise, they should not be trying to require `fence-agents-all`.

3. In checking on Rocky 8, I found that there is nowhere that the `fences-agents` package exists, but that even then, it had been renamed to `fence-agents-all`, but was still in [the 8.3 AppStream repo](https://dl.rockylinux.org/vault/rocky/8.3/AppStream/x86_64/os/Packages/) as of at least 8.3. So no tricks needed to enable a repository. I wonder why it was not breaking...

For now, I am going to focus on the files for Rocky 9, and come back to the broken workflow items later.

**Reply by SchoolGuy** [reply]
In regard to 2. I can help you: We are using (only in `release33`) the program `debbuild` which uses the specfile to build the DEB packages. As such you will find `%if` followed by Debian logic.

**Reply by cinnion** [reply]
Thanks @SchoolGuy!  After some research, I found `debbuild` has some issues I nearly exercised, but turns out I flipped my `%if` logic apparently. Grr.  Anyways, I have another commit I am going to be pushing here  shortly (as soon as one last local run of the workflows using `nektos/act` completes after a few last tweaks). I fixed the name for the Rocky 8 workflow, changing it from `build-centos8-rpms` to `build-rocky8-rpms` since CentOS != Rocky these days, and added one for Rocky 9, which has a dockerfile which works for it (a few packages are no longer distributed, like Grub EFI for i686, others got moved to EPEL and renamed). But it seems to be passing locally, and I want to see how things go for the real workflow executor, especially since some workflows fail locally whether we are talking on the unmodified `release33` branch or my fix branch.  My though it to also eventually add something about `nektos/act` to the developer docs, as it seems to allow us to run the workflows locally.

**Reply by cinnion** [reply]
@SchoolGuy, on the pull request, you said "You didn't add a fragment.", but fixed it. Was all of the commit you did to fix the fragment itself, or just things in general that I had missed (or did not do, like renaming from centos8 to rocky8, in my thinking to keep changes to a minimum)? I want to do better on my next PR. :D

Now, if I can only get my `cobbler import` issue resolved, so that I can retire my 2.8.3 install, I will see about giving koan a little bit bit of love.

**Reply by SchoolGuy** [reply]
So I did a few things:
- Add the changelog Fragment
- Fix the Makefile
- Added the symlink for the system-test that you added the index for

Maybe I did more but that should be the most important things. You did good and Cobbler is one of the more weird/complex projects, so don't be too hard on yourself.

From a personal desire the only thing I would have liked is if you would have created the devel version of the PR first and then the backport. This was rather trivial and forward porting this was literally a cherry-pick and four lines of adjustments but the other way around is normally easier.

**Reply by SchoolGuy** [reply]
I'll close this discussion since the contribution for this topic is completed. Thanks again!

**Reply by cinnion** [reply]
OK. Got it on the devel next time and backport. Contrary to almost all other projects I have worked on, cobbler is backwards in that main (or master) has typically been what went to production, not for the development. I will try to remember that fact here. :)
